File: assets/js/lib/issue.js


class Issue {
    
    constructor(data) {
        this.id = parseInt(data.id);
        this.parent = parseInt(data.parent);
        this.name = data.name;
        this.description = data.description;
        this.project = parseInt(data.project);
        this.type = data.type.toLowerCase();
        this.userCreated = parseInt(data.userCreated);
        this.userAssigned = parseInt(data.userAssigned);
        this.state = parseInt(data.state);
        this.position = data.position.toLowerCase();
    }
}

File: lib/issue_tracker/models/issue.rb
require 'dm-core'
require 'dm-migrations'

class Issue

  include DataMapper::Resource

  property :id, Serial
  property :parent, Integer, required: true
  property :name, String, required: true
  property :description, String
  property :user_created, Integer, required: true
  property :user_assigned, Integer, required: true

  belongs_to :type
  belongs_to :project
  belongs_to :state
  belongs_to :sprint

end